snuba/pipeline/pipeline_delegator.py

```diff
@@ -26,6 +29,20 @@
 ]
 
 
+def _is_query_copying_disallowed(current_referrer: str) -> bool:
+    """
+    Check if the given referrer is disallowed from making a copy of the query.
```

Comment:
maybe you could add why a given referrer would be disallowed from making a copy of the query? I'm not sure it's fully obvious to me why subscriptions and subscriptions executor are too sensitive for using deepcopy